### DISCUSSION TITLE
BLE CharacteristicInfo Discovery

### DISCUSSION THREAD

**Discussion by RobSchein** [shinyorg/shiny#1461] [Q&A]
Hi Allan.  Made a ton of progress with Shiny BLE since my first question.  Thanks a ton.  Refactoring code now.

Question on...
var CharacteristicListReadOnly = await Peripheral.GetCharacteristicsAsync(ServiceInfo.Uuid);

The list that is retrieved contains all the characteristics in my BLE peripheral associated with several services and the peripheral itself, and not just the ones associated with that ServiceInfo UUID.  I am manually filtering to retrieve the ones of interest.  Just wondering whether GetCharacteristicsAsync(ServiceInfo.Uuid) discovery this way should be doing the filtering?  What am I missing.

Thanks in advance,
-Rob

**Comment by aritchie** [comment]
Issues/Enhancement Requests go here so I don't have to ask the same questions every time: https://github.com/shinyorg/shiny/issues/new/choose
